client/src/components/DatasetInformation/DatasetAttributes.test.ts

```diff
@@ -1,58 +1,70 @@
+async function montDatasetAttributes(conversion_disable = false) {
+    const pinia = createTestingPinia();
+    setActivePinia(pinia);
+
+    const axiosMock = new MockAdapter(axios);
+    axiosMock.onPut(`/dataset/set_edit`).reply(200, { message: "success", status: "success" });
+    axiosMock.onGet(`/dataset/get_edit?dataset_id=${DATASET_ID}`).reply(200, {
+        attribute_inputs: [{ name: "attribute_text", type: "text" }],
+        conversion_inputs: [{ name: "conversion_text", type: "text" }],
+        conversion_disable: conversion_disable,
+        datatype_inputs: [{ name: "datatype_text", type: "text" }],
+        permission_inputs: [{ name: "permission_text", type: "text" }],
+    });
+
+    const wrapper = mount(DatasetAttributes as object, {
         propsData: {
-            datasetId: "dataset_id",
+            datasetId: DATASET_ID,
         },
         localVue,
-        stubs: {
-            DatasetAttributesProvider: MockProvider({
-                result: {
-                    attribute_inputs: [{ name: "attribute_text", type: "text" }],
-                    conversion_inputs: [{ name: "conversion_text", type: "text" }],
-                    conversion_disable: conversion_disable,
-                    datatype_inputs: [{ name: "datatype_text", type: "text" }],
-                    permission_inputs: [{ name: "permission_text", type: "text" }],
-                },
-            }),
-            FontAwesomeIcon: false,
-            FormElement: false,
-        },
+        pinia,
     });
+
     await flushPromises();
+
     return wrapper;
 }
 
 describe("DatasetAttributes", () => {
     it("check rendering", async () => {
-        const axiosMock = new MockAdapter(axios);
-        axiosMock.onPut(`/dataset/set_edit`).reply(200, { message: "success", status: "success" });
-        const wrapper = await buildWrapper();
+        const wrapper = await montDatasetAttributes();
+
         expect(wrapper.findAll("button").length).toBe(6);
         expect(wrapper.findAll("#attribute_text").length).toBe(1);
         expect(wrapper.findAll("#conversion_text").length).toBe(1);
         expect(wrapper.findAll("#datatype_text").length).toBe(1);
         expect(wrapper.findAll("#permission_text").length).toBe(1);
         expect(wrapper.findAll(".tab-pane").length).toBe(3);
         expect(wrapper.findAll(".ui-portlet-section").length).toBe(2);
-        const $button = wrapper.find("#dataset-attributes-default-save");
-        await $button.trigger("click");
+
+        const saveButton = wrapper.find("#dataset-attributes-default-save");
+
+        await saveButton.trigger("click");
+
         await flushPromises();
+
         expect(wrapper.find("[role=alert]").text()).toBe("success");
     });
 
     it("check rendering without conversion option", async () => {
-        const wrapper = await buildWrapper(true);
+        const wrapper = await montDatasetAttributes(true);
+
+        await flushPromises();
```

Comment:
When a button is clicked, an asynchronous function is triggered and it waits for the result to be returned.

Reply:
Promises should be already ready at this point, because it is called at the end of the `mountDatasetAttributes` function, there seems to be no button trigger here.

Also when there is a button trigger, calling `await saveButton.trigger("click");` will be enough I think :thinking:

Reply:
Oh, I was wrong. you are right. Thank you!

Reply:
For a button call, at least here, it's needed to call flushPromises.